Cython/Compiler/ExprNodes.py

```diff
@@ -723,6 +744,10 @@ def allocate_temp_result(self, code):
                 return
             self.temp_code = code.funcstate.allocate_temp(
                 type, manage_ref=self.use_managed_ref)
+
+            if self.type.is_cpp_class:
+                code.globalstate.use_utility_code(
+                    UtilityCode.load_cached("MoveIfSupported", "CppSupport.cpp"))
```

Comment:
Why not load the utility code in `_make_move_result_rhs()`? Just pass in `code`.

Reply:
It would force me to change the interface of `calculate_result_code` because that doesn't have `code`.

I think the place to put it might be in `generate_disposal_code` since at that stage we have the code and know if the temp was moved (but this does relying on it anyone who overrides it remembering, and the disposal code being called correctly)

Reply:
> [depends on] the disposal code being called correctly

Well, if that's not done correctly then that's a bug, but a different one, and not something we should care about here.

What about `generate_post_assignment_code()` ? Isn't that impacted, too?

Reply:
Yes I think it does need to be in `generate_post_assignment_code` too. I've added that